cmake/reports/CMakeLists.txt

```diff
@@ -3,7 +3,11 @@
 set(flag_for_ram_report ram)
 set(flag_for_rom_report rom)
 set(flag_for_footprint all -q)
-set(report_depth 99)
+
+set(report_depth 99 CACHE STRING "Depth of tree to display")
+set(sort_key "name" CACHE STRING "Sort key for results display (name or size)")
+set_property(CACHE sort_key PROPERTY STRINGS "name" "size")
+
```

Comment:
please use a `zephyr_get(FOOTPRINT_REPORT_DEPTH)` for this variable and related ones. (if of course CMake variables are agreed to be used.)

This ensures a uniform way of getting CMake variables in Zephyr, as well as include support for variables passed through sysbuild .

See more here: https://github.com/zephyrproject-rtos/zephyr/blob/8d6b206064c915316245616a718ae29eb157a717/cmake/modules/extensions.cmake#L2469-L2497